test/performance/operf.go

```diff
@@ -0,0 +1,666 @@
+)
+
+type Metric struct {
+	Min float64
+	Max float64
+	Avg float64
+}
+
+type ObjectMetric struct {
+	Name string
+	Type string
+	Cpu  Metric
+	Mem  Metric
+}
+
+type LvmUnitMetrics struct {
+	Start         time.Time
+	End           time.Time
+	Name          string
+	Namespace     string
+	WorkloadType  string
+	ObjectMetrics map[string]ObjectMetric
+}
+
+const (
+	CpuQueryTemplatePod       = `sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{cluster="", namespace="{{.Namespace}}"} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	CpuQueryTemplateContainer = `sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{cluster="", namespace="{{.Namespace}}"} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (container)`
+	MemQueryTemplatePod       = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	MemQueryTemplateContainer = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (container)`
+	Mebibyte                  = 1048576 // bytes
+)
+
+const (
+	UsageHelp = `Usage of operf:
+This script provides LVM operator metrics retrieval (for all the units in the operator):
+
+- PVCs creation and usage:
+	It creates <instances> of "busybox" pods using PVCs provisioned via the Storage class provided
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -instances 16
+
+Parameters:`
+)
+
+var (
+	// flags
+	Instances            *int
+	TestNamespace        *string
+	TestStorageClassName *string
+	NamePattern          *string
+	Token                *string
+
+	// variables
+	PrometheusURL string
+	PVCLabels     = map[string]string{"app": "testPerformance"}
+	Ctx           = context.Background()
+)
+
+func main() {
+	Instances = flag.Int("instances", 4, "Number of Pods/Pvcs (each Pod uses a different PVC) to create in the test")
+	TestNamespace = flag.String("namespace", "lvm-operator-system", "Namespace where operator is deployed and the PVCs and test pods will be deployed/undeployed")
+	TestStorageClassName = flag.String("sc", "odf-lvm-vg1", "Name of the topolvm storage class that will be used in the PVCs")
```

Comment:
`sc` should also be made mandatory IMO.  Using a default value of `odf-lvm-vg1` might not work every time.

Reply:
Why it won't work, We always create the SC with this name right? If in some cases it does not work they can still pass it.

Reply:
The sc name depends on the name of the VG in the LVMCluster CR. It would be better to require the sc name to be provided as well as there is not guarantee that it would be odf-lvm-vg1.

Reply:
I see